lib/ace/ext/papeeria/katex-previewer.coffee

```diff
@@ -126,16 +138,16 @@ define((require, exports, module) ->
         @lastChangeTime = null
       else
         @currentDelayedUpdateId = null
-        { row: cursorRow, column: cursorColumn } = @editor.getCursorPosition()
-        curContext = LatexParsingContext.getContext(@editor.getSession(), cursorRow, cursorColumn)
+        cursorPos = @editor.getCursorPosition()
+        curContext = LatexParsingContext.getContext(@editor.getSession(), cursorPos.row, cursorPos.column)
 
         if curContext == "equation"
           @updateRange()
         else
           @destroyRange()
           @disableUpdates()
 
-        if @rangeExists and @rangeCorrect
+        if @currentRange?
```

Comment:
is it okay that you don't check `rangeCorrect` anymore? It is possible that currentRange is defined while `rangeCorrect` is still false, right? The same question about a copule of similar replaces below.